domain/machine/state/machine_cloud_instance.go

```diff
@@ -83,23 +86,52 @@ VALUES ($instanceTag.*)
 		return errors.Trace(err)
 	}
 
+	azName := availabilityZoneName{}
+	if hardwareCharacteristics != nil && hardwareCharacteristics.AvailabilityZone != nil {
+		az := *hardwareCharacteristics.AvailabilityZone
+		azName = availabilityZoneName{Name: az}
+	}
+	retrieveAZUUID := `
+SELECT uuid AS &availabilityZoneName.uuid
```

Comment:
`SELECT &availabilityZoneName.uuid`

Reply:
Nice I didn't know this worked, can use it in my current PR.

Reply:
Yeah nice! Thanks Joe, I'll update.